projects/igniteui-angular/src/lib/paginator/paginator.component.ts

```diff
@@ -0,0 +1,247 @@
+import { CommonModule } from '@angular/common';
+import { FormsModule } from '@angular/forms';
+import { Component, Input, Output, NgModule, Optional, Inject, EventEmitter, HostBinding } from '@angular/core';
+import { CurrentResourceStrings } from '../core/i18n/resources';
+import { IDisplayDensityOptions, DisplayDensityToken, DisplayDensityBase, DisplayDensity } from '../core/displayDensity';
+import { IgxSelectModule } from '../select/index';
+import { IgxIconModule } from '../icon/index';
+import { IgxButtonModule } from '../directives/button/button.directive';
+import { IgxRippleModule } from '../directives/ripple/ripple.directive';
+
+@Component({
+    selector: 'igx-paginator',
+    templateUrl: 'paginator.component.html',
+})
+export class IgxPaginatorComponent extends DisplayDensityBase {
+
+    /**
+     * Total pages calculated from totalRecords and perPage
+     */
+    public totalPages: number;
+    protected _page = 0;
+    protected _totalRecords: number;
+    protected _selectOptions = [5, 10, 15, 25, 50, 100, 500];
+    protected _perPage = 15;
+
+    /**
+     * @hidden
+     */
+    @HostBinding('class')
+    public get classes(): string {
+        switch (this.displayDensity) {
+            case DisplayDensity.cosy:
+                return 'igx-grid-paginator--cosy';
+            case DisplayDensity.compact:
+                return 'igx-grid-paginator--compact';
+            default:
+                return 'igx-grid-paginator';
+        }
+    }
+
+    /**
+     * @hidden
```

Comment:
This should not be a hidden input. It needs to be documented.

Reply:
Btw, the way this is now bound means that if the user assigns a custom class to the igx-paginator their class won't stick and will be overwritten by the classes method.